frontend/src/views/SecretOverviewPage/SecretOverviewPage.tsx

```diff
@@ -183,6 +185,41 @@ export const SecretOverviewPage = () => {
     environments: userAvailableEnvs.map(({ slug }) => slug)
   });
 
+  const selectAllEntries = () => {
+    const folderRecord: Record<string, boolean> = {};
+    folderNames.forEach((folder: string) => {
+      folderRecord[folder] = true;
+    })
+
+    const secretRecord: Record<string, boolean> = {};
+    secKeys.forEach((secret: string) => {
+      secretRecord[secret] = true;
+    })
+
+    setSelectedEntries({
+      [EntryType.FOLDER]: folderRecord,
+      [EntryType.SECRET]: secretRecord
+    })
+  };
+
+  const allSelections = useMemo(() => {
```

Comment:
Can't we just make it into one useMemo rather than 3

Reply:
There two areas to click on the folder row:
* anywhere except the icons: there will be an route change and on any route change, the store will have an reset. 
* icons (folder icon, edit icon, delete icon): there wouldn't be any reset.

In the above comment: Do you mean that when clicking on the folder icon, there is no reset?
I think no reset on the folder icon click signifies that the folder cannot be selected on the environment page.
Reset on folder icon click might confuse the user.

Please let me know if I misunderstood your comment 🫣

Reply:
Removed the use of multiple useMemo in 470d0827e5cada3facd0a322a90c3e066e9357a5